newIDE/app/src/EventsFunctionsList/ObjectTreeViewItemContent.js

```diff
@@ -0,0 +1,296 @@
+// @flow
+import { type I18n as I18nType } from '@lingui/core';
+import { t } from '@lingui/macro';
+
+import * as React from 'react';
+import newNameGenerator from '../Utils/NewNameGenerator';
+import Clipboard, { SafeExtractor } from '../Utils/Clipboard';
+import {
+  serializeToJSObject,
+  unserializeFromJSObject,
+} from '../Utils/Serializer';
+import { type HTMLDataset } from '../Utils/HTMLDataset';
+import {
+  TreeViewItemContent,
+  type TreeItemProps,
+  extensionObjectsRootFolderId,
+} from '.';
+import { getShortcutDisplayName } from '../KeyboardShortcuts';
+import defaultShortcuts from '../KeyboardShortcuts/DefaultShortcuts';
+
+const EVENTS_BASED_OBJECT_CLIPBOARD_KIND = 'Events Based Object';
+
+export type EventObjectCallbacks = {|
+  onSelectEventsBasedObject: (eventsBasedObject: ?gdEventsBasedObject) => void,
+  onDeleteEventsBasedObject: (
+    eventsBasedObject: gdEventsBasedObject,
+    cb: (boolean) => void
+  ) => void,
+  onRenameEventsBasedObject: (
+    eventsBasedObject: gdEventsBasedObject,
+    newName: string,
+    cb: (boolean) => void
+  ) => void,
+  onEventsBasedObjectRenamed: (eventsBasedObject: gdEventsBasedObject) => void,
+|};
+
+export type EventObjectProps = {|
+  ...TreeItemProps,
+  ...EventObjectCallbacks,
```

Comment:
I know it's verbose, but I think it's worth avoiding "EventObject" and keeping "EventsBasedObject" everywhere. If anything, this helps to grep the codebase for it.

```suggestion
export type EventsBasedObjectCallbacks = {|
  onSelectEventsBasedObject: (eventsBasedObject: ?gdEventsBasedObject) => void,
  onDeleteEventsBasedObject: (
    eventsBasedObject: gdEventsBasedObject,
    cb: (boolean) => void
  ) => void,
  onRenameEventsBasedObject: (
    eventsBasedObject: gdEventsBasedObject,
    newName: string,
    cb: (boolean) => void
  ) => void,
  onEventsBasedObjectRenamed: (eventsBasedObject: gdEventsBasedObject) => void,
|};

export type EventsBasedObjectProps = {|
  ...TreeItemProps,
  ...EventsBasedObjectCallbacks,
```